src/Apps/Order/Routes/Shipping2/ShippingContext.tsx

```diff
@@ -0,0 +1,176 @@
+import { createContext, FC } from "react"
+import {
+  ComputedOrderData,
+  computeOrderData,
+} from "Apps/Order/Routes/Shipping2/Utils/computeOrderData"
+import { useReducer } from "react"
+import { ShippingProps, ShippingStage } from "Apps/Order/Routes/Shipping2"
+import { FormikProps } from "formik"
+import { FulfillmentValues } from "Apps/Order/Routes/Shipping2/Utils/shippingUtils"
+import createLogger from "Utils/logger"
+import { Dialog } from "Apps/Order/Dialogs"
+import { useHandleExchangeError } from "Apps/Order/Routes/Shipping2/Hooks/useHandleExchangeError"
+
+export interface State {
+  formHelpers: FormHelpers
+  newSavedAddressId: string | null
+  selectedShippingQuoteId: string | null
+  stage: ShippingStage
+  isPerformingOperation: boolean
+  isArtsyShipping: boolean
+}
+
+interface Actions {
+  dialog: Dialog
+  handleExchangeError: (
+    error: {
+      code: string
+      data: string | null | undefined
+    },
+    logger: ReturnType<typeof createLogger>
+  ) => void
+  setFormHelpers: (payload: FormHelpers) => void
+  setSelectedShippingQuote: (payload: string | null) => void
+  setNewSavedAddressId: (payload: string | null) => void
+  setIsPerformingOperation: (payload: boolean) => void
+  setStage: (payload: ShippingStage) => void
+}
+
+export interface ShippingContextProps {
+  actions: Actions
+  state: State
+  orderData: ComputedOrderData
+}
+
+export const ShippingContext = createContext<ShippingContextProps>({} as any)
+
+export const ShippingContextProvider: FC<Pick<
```

Comment:
Now with all of the behavior co-located where it needs to live, the context is quite slim